synapse/storage/databases/main/events_worker.py

```diff
@@ -525,6 +526,57 @@ def _get_events_from_cache(self, events, allow_rejected, update_metrics=True):
 
         return event_map
 
+    async def get_stripped_room_state_from_event_context(
+        self,
+        context: EventContext,
+        state_types_to_include: List[EventTypes],
+        membership_user_id: Optional[str],
+    ) -> List[JsonDict]:
+        """
+        Retrieve the stripped state from a room, given an event context to retrieve state
+        from as well as the state types to include. Optionally additionally include membership
+        events from a specific user.
```

Comment:
The "Optionally additionally" is a bit confusing.

```suggestion
        from as well as the state types to include. Optionally, include the membership
        events from a specific user.
```